### PR TITLE
Improve Windows Event Log mapping for collector input

### LINKED ISSUE #25161: Check Collector windows event log GIM mapping
Check all fields for the windows event log -> GIM mapping.

### PR DESCRIPTION
**Pull request by kroepke**
/nocl unreleased feature

Fixes #25161

## Summary

Overhauls the `WindowsEventLogRecordProcessor` to produce a robust, vendor-neutral baseline of Windows Event Log data for the OTLP collector input. System-level and unambiguous EventData fields are normalized to GIM; user-identity fields whose meaning varies per event ID are passed through with their raw MS XML attribute names for a per-event-aware layer (Illuminate) to finish.

## Rationale

Windows events reuse the Subject/Target terminology with different meanings per event ID:

- For **4624** (Logon), Subject is the local service requesting the logon; Target is the actual human authenticating.
- For **4656** (Handle requested), Subject is the acting user and there is no Target.
- For **4672** (Special privileges), Subject is the sole user receiving privileges; there is no Target.
- For **4688** (Process creation), Subject is the creator and Target is the new-process owner — potentially a different user.

Mapping Subject/Target into `user_*` / `source_user_*` without knowing the event ID is wrong for some events, and committing to an interpretation in 8.0 would force a breaking change the first time we correct it. The processor therefore stays neutral on these specific fields and lets the per-event-aware layer (Illuminate) promote them to the appropriate GIM user buckets. This is the same division of labor that already exists for Winlogbeat and NXLog inputs.

## What the processor does

- **Normalizes unambiguous fields to GIM**: event metadata (`event_code`, `event_source`, `event_log_name`, `event_uid`, `vendor_event_timestamp`, `vendor_subtype`, `vendor_event_category`, `vendor_opcode`, `vendor_version`, `vendor_event_severity`, `event_outcome`, `trace_id`), host (`host_hostname`), process (`process_id`, `process_name`, `process_path`, `process_parent_*`, `process_command_line`), network source (`source_ip`, `source_port`, `source_hostname`), destination (`destination_hostname`), auth/logon context (`windows_logon_type`, `windows_authentication_*`), error (`event_error_code`, `event_error_description`), privileges (`privilege_assigned_name`).
- **Passes through raw** (event-ID-conditional meaning): `SubjectUserSid`, `SubjectUserName`, `SubjectDomainName`, `SubjectLogonId`, `TargetUserSid`, `TargetUserName`, `TargetDomainName`, `TargetLogonId`, `TargetAccount`, `LogonGuid`, `SubStatus`.
- **Preserves everything else in `vendor_event_data`**: the complete original EventData/UserData structure is serialized as a JSON blob so nothing is lost, regardless of what the flat fields capture.
- **Handles all channels** (Security, System, Application, PowerShell, Defender, OpenSSH, etc.), not just Security.

## Follow-on work

A parallel Illuminate content pack will be added to promote the raw user-identity fields to `user_*` / `source_user_*` / `target_user_*` per event ID, mirroring the per-event rules that already exist for Winlogbeat and NXLog inputs.

## Breaking changes

- `associated_user_*` fields (previously produced by this processor) are no longer emitted.
- For users who consumed the interim `source_user_*` / `user_*` mapping that existed earlier on this branch, those fields are also no longer emitted directly by the processor — they will be re-introduced by Illuminate content per-event. Saved searches, dashboards, or pipeline rules keyed on those field names for collector-input data will need to wait for the Illuminate follow-on, or read from the raw `SubjectUserName`/`TargetUserName`/etc. fields in the meantime.

## Test Plan

- [x] 20 unit tests covering field mappings, raw passthrough, JSON blob serialization, and edge cases (UserData-only events, Security-section-only events, PowerShell, Defender, OpenSSH)
- [x] Fixtures include real OTLP records for events 4624, 4625, 4634, 4648, 4672, 4688, 4717, 104, Defender 1150, System 7036
- [x] Live validation against replayed OTLP traffic for 4624, 4625, 4634, 4672 — raw field layout verified, `vendor_event_data` blob preserves full original structure, non-user Illuminate rules (IP processing, hostname enrichment) continue to fire correctly

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/collectors/input/processor/WindowsEventLogRecordProcessor.java [hunk 1]
```diff
@@ -509,12 +521,59 @@ private static List<String> toMultivalue(String value) {
         return values;
     }
 
+    private static String serializeKvListToJson(List<KeyValue> kvList) {
```
Comment (thll) [inline graylog2-server/src/main/java/org/graylog/collectors/input/processor/WindowsEventLogRecordProcessor.java:524]: Can we maybe reuse existing utilities here instead of reimplementing? See e.g. https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog/inputs/otel/codec/OTelLogsCodec.java#L187

File: graylog2-server/src/main/java/org/graylog/collectors/input/processor/WindowsEventLogRecordProcessor.java [hunk 2]
```diff
@@ -43,19 +44,36 @@
...
+ *
+ * <p>All EventData/UserData entries are additionally serialized into {@code vendor_event_data}
+ * as a JSON blob to preserve the complete original structure for downstream processing.
  *
  * @see <a href="https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/windowseventlogreceiver">windowseventlog receiver</a>
  * @see <a href="https://learn.microsoft.com/en-us/windows/win32/wes/eventschema-systempropertiestype-complextype">SystemPropertiesType</a>
  */
 public class WindowsEventLogRecordProcessor implements LogRecordProcessor {
-    private static final String EVENT_ID = "event_id";
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
```
Comment (thll) [inline graylog2-server/src/main/java/org/graylog/collectors/input/processor/WindowsEventLogRecordProcessor.java:68]: Is there a reason why we use a separate object mapper?

**Review by thll — CHANGES_REQUESTED** [review]

File: graylog2-server/src/main/java/org/graylog/collectors/input/processor/WindowsEventLogRecordProcessor.java (hunk 2 above)
Comment (kroepke) [reply graylog2-server/src/main/java/org/graylog/collectors/input/processor/WindowsEventLogRecordProcessor.java]: replaced it with the otel type mapper, it was an oversight.

**Review by thll — APPROVED** [review]
